hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/HazelcastCacheRegionFactory.java

```diff
@@ -58,6 +56,14 @@ public EntityRegion buildEntityRegion(final String regionName, final Properties
                 new IMapRegionCache(regionName, instance, properties, metadata));
     }
 
+    public NaturalIdRegion buildNaturalIdRegion(final String regionName, final Properties properties,
+                                                final CacheDataDescription metadata)
+            throws CacheException {
+        /* Natural id regions are never versioned, so pass in null for metadata */
+        return new HazelcastNaturalIdRegion<IMapRegionCache>(instance, regionName, properties, metadata,
+                new IMapRegionCache(regionName, instance, properties, null));
```

Comment:
Please see my note "The comment `/* Natural id regions are never versioned, so pass in null for metadata */` above explains why." above